File: Data Structures & Algorithms/two-integer-sum/submission-5.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        for(int i = 0; i<nums.size(); i++){
            mp[nums[i]] = i+1;
        }
        for(int i = 0; i<nums.size(); i++){
            int diff = target - nums[i];
            if(mp[diff] && mp[diff]!= i+1){
                return {i,mp[diff] - 1};
            }
        }


    }
};

File: Data Structures & Algorithms/binary-search/submission-5.java
class Solution {
    public int search(int[] nums, int target) {
        return helper(nums, target, 0, nums.length-1);
    }

    public int helper(int[] nums, int target, int l, int r) {
        if (l > r) {
            return - 1;
        }
        int m = (l+r)/2;
        
        if(nums[m] == target) {
            return m;
        }
        if(nums[m] > target) {
            return helper(nums, target, l, m-1);
        }
        else {
            return helper(nums, target, m+1, r);
        }
    }

}
